### DISCUSSION TITLE
For Jumper T18, add a bin file to flash firmware version 4192 to GT911.

### DISCUSSION THREAD

**Discussion by 1979mgb** [EdgeTX/edgetx#3138] [Ideas]
Jumper T18's have socket to add touch screen. After installing a touch screen, in Edge TX I went to SYS/Hardware. Then scrolling down to Debug and touching Analogs a line at bottom of screen states:
Touch GT911 FW ver: 0
On my Radiomaster TX16S the same screen states:
Touch GT911 FW ver: 4192
Can a bin file be created to flash the Touch GT911 installed in the Jumper T18 to upgrade the firmware to ver: 4192?

**Comment by pfeerick** [comment]
@rotorman Do you know if we are able to update the firmware on the GT911? But before that, I seem to think the T18 needs resistors removed for the touch panel to work?

**Reply by rotorman** [reply]
GT911 firmware should be updated automatically, if a newer version number is listed in configuration that presently loaded: https://github.com/EdgeTX/edgetx/blob/6f4abef36a9dbd4bafca44396b08e01868cb617d/radio/src/targets/horus/tp_gt911.h#L45

If it still reads out 0, then likely communication with GT911 is not working (yet). Please see: https://github.com/EdgeTX/edgetx/issues/803#issuecomment-927408201 on how two resistors on T18/TX18S mainboard need to be removed for touch to work.

(The GT911 configuration value can only be incremented higher. To get back to 0 e.g. a CLI command, as described in: https://github.com/EdgeTX/edgetx/pull/2351, can be used. This is relevant only though, if one has experimented with GT911_CFG_NUMBER and should be irrelevant here.)

**Comment by 1979mgb** [comment]
Thank you all for your help. Hope to try these today.
Charles

**Comment by 1979mgb** [comment]
Hello Peter, Thank you for reply,
Sounds like No one has tried this in the Jumper T18 yet. I have followed
all posts on the internet and can only find Mods to the T16 which is
totally different. It did not have a connector on the main board.
The T18 has a dedicated socket for the ribbon cable of the touch screen.
Either way if 2 resistors need to be removed, they are probably different
from the ones removed on the T16.
Charles

**Comment by 1979mgb** [comment]
Hello Risto,
Thank you for your reply.
I followed the link to #2060 which discusses modifying a T16. The T18 has a
dedicated ribbon connector on the main board. Do the 2 resistors on the T18
need to be removed? Any additional wiring needed?
Thank you,
Charles

**Comment by rotorman** [comment]
Wrong link, my bad. See here: https://github.com/EdgeTX/edgetx/issues/803#issuecomment-927408201 (I updated the original post of mine above as well, in case someone else stumbles upon this discussion).

**Comment by 1979mgb** [comment]
Risto,
Thank you so much, didn't want to try removing resistors by using the T16
photos.
Will give this a go.
Charles

**Comment by pfeerick** [comment]
This is what I was thinking of... from one of the OTX devs... I think there should be some discussion on RCGroups about the T18 and required mod if you check there. edit: Ok, I see the photo showing the two resistors needing to be removed is on Risto's second reply.

![image](https://user-images.githubusercontent.com/5500713/216512829-81662479-178f-48d5-ac61-fc7e8b4a7f85.png)

**Comment by 1979mgb** [comment]
Good afternoon,
Thank you all for helping me find the information I needed.
I removed the 2 resistors as shown in #803, reinstalled my battery, turned on the radio and touch is working. 
Firmware version 4192 is now showing for the GT911.
I purchased my touch screen from Aliexpress for $19.95 US.
The only other thing that is needed for this mod is to trim plastic from the screen holder to allow for the extra thickness of the touch plate.
I searched the web for how to do this mod, all results came to the T16 modification. Even here on Github a search for Jumper T18 on "all Github" comes up with no results.     
So, a very big thank you to all the contributors.
Charles
